GlobalConstantsService/inc/ParticleData.hh

```diff
@@ -0,0 +1,63 @@
+#ifndef GlobalConstantsService_ParticleData_hh
+#define GlobalConstantsService_ParticleData_hh
+//
+// Partcile data loaded from ParticleDataList
+//
+// the content is
+//  - PDG ID
+//  - displayname (includes special chars like +-*)
+//  - progamming name (includes only alphanumeric and _)
+//        this should be the same as  DataProducts/inc/PDGCode.hh
+//  - charge (units of electron charge)
+//  - mass (MeV)
+//  - lifetime (ns)
+//
+
+#include <iostream>
+#include <iomanip>
+
+#include "Offline/GlobalConstantsService/inc/ParticleData.hh"
+
+namespace mu2e {
+
+
+  class ParticleData {
+
+  public:
+
+    ParticleData(int id, const std::string& name, const std::string& codeName,
+                 float charge, float mass, float lifetime): 
+      _id(id),_name(name),_codeName(codeName),_charge(charge),
+      _mass(mass),_lifetime(lifetime) {}
+
+    int id() const { return _id; }
+    const std::string& name() const { return _name; }
+    const std::string& displayName() const { return _name; }
+    const std::string& codeName() const { return _codeName; }
+    float charge() const { return _charge; }
+    float mass() const { return _mass; }
+    float lifetime() const { return _lifetime; }
+
+    void print( std::ostream& ostr=std::cout) const {
+      ostr << std::setw(11) << _id;
+      ostr << std::setw(22) << _name;
+      ostr << std::setw(35) << _codeName;
+      ostr << std::setw(10) << _charge;
+      ostr << std::setw(12) << _mass;
+      ostr << std::setw(12) << _lifetime;
+      ostr << std::endl;
+    }
```

Comment:
I suggest writing operator << and then implementing the print function using that.   Also I would put the implementation of operator << in the .cc file since I think that the code ends up being biggish to inline.

I know that I started this pattern but I would like to modernize it.

Reply:
Thanks for doing this.  One last request.  If operator << is implemented using the public interface, then you do not need to friend it.  I believe everything that you do print is in the public interface.

Reply:
Thanks.